elementary/monitor/api/tests/tests.py

```diff
@@ -202,7 +195,15 @@ def get_test_results(
             test_results_metadata = self.get_tests_metadata(days_back=days_back)
 
         tests_sample_data = self.get_run_cache(TESTS_SAMPLE_DATA)
-        if tests_sample_data is None:
+        if invocation.invocation_id:
```

Comment:
Get only tests samples for the filtered invocation and not all the latest test runs (improve report performances)